langkit/metrics/injections.py

```diff
@@ -0,0 +1,87 @@
+import os
+from functools import partial
+from typing import Any, Sequence
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import torch
+from sentence_transformers import SentenceTransformer
+
+from langkit.core.metric import Metric, SingleMetric, SingleMetricResult
+from langkit.metrics.util import LazyInit, get_data_home
+
+__transformer_name = "all-MiniLM-L6-v2"
+__version = "v2"
+__injections_base_url = "https://whylabs-public.s3.us-west-2.amazonaws.com/langkit/data/injections/"
+__embeddings = LazyInit(lambda: __load_embeddings())
+__transformer = LazyInit(
+    lambda: SentenceTransformer(__transformer_name, device="cuda" if torch.cuda.is_available() else "cpu")
+)
```

Comment:
I think it's best not to couple these - we currently use the same embedding model for every module, but it's possible that different models might be more well-suited for different metrics.

Reply:
I think the important part is knowing if its ok to have multiple instances (will each one take up several hundred mb of ram, for example). The coupling will be minimal and only known to us, but the resource consumption will be felt by every user.

Do you know what the deal is here, resource wise?

Reply:
the model we're using is ~90MB. I guess we can share the same model between metrics. If we decide to change the model for one module, we then would a) have to decouple it from other metrics, or b) change all of them at once. b) could be trickier, because, for the injections module, this means having to regenerate artifacts at WhyLabs' S3

Reply:
The LazyInit class handles this pretty gracefully. It'| a simple as moving that out into a file where they can both reference it as a `LazyInit[SentenceTransformer]`. We would just stop referencing that variable, no refactoring needed and no trickiness.

Reply:
ok, I put it under `transformer.py` to make it accessible from other metrics (currently injections and input_output). Can you give it another look?